### DISCUSSION TITLE
Support passing predefined body and header parameters in the `open` method returned by `createApiClientModal`

### DISCUSSION THREAD

**Discussion by TimoGlastra** [scalar/scalar#2528] [Ideas]
We are looking into the play button component and integrate it into the core of our product, to easily let users interact with the API. 

The Play button seems like a good fit for this, in addition to the API reference we already host (http://paradym.id/reference). 

It would be really useful if the `open()` method returned by `createApiClientModal` would also allow to pass in a pre-filled body as well as variables for request headers, etc..

This would allow us to pre-populate the play button based on the page where the user is currently at. See for example the design below, where we show an example payload for the API, and we have a "Use in reference" button that links to the reference. If, instead we could use the play button (or a custom button) and load the API client modal, and open with all parameters pre-filled that would be really useful.

![image (4)](https://github.com/user-attachments/assets/1868f867-f3f0-4621-be9c-48321150ad50)

I think as a workaround for now we could on-demand generate the open api spec and populate the example values with the actual data, so it will be pre-filled.

Maybe it would be partially solved by https://github.com/scalar/scalar/pull/2445, as we could provide the variables for replacement?

**Comment by hanspagel** [comment]
This looks rad! ~What are you building?~ Ah, saw the link!

Love that idea, we should definitely add it! Yes, the quick workaround would be to pass variables.
